azurerm/resource_arm_redis_cache.go

```diff
@@ -241,7 +248,7 @@ func resourceArmRedisCacheCreate(d *schema.ResourceData, meta interface{}) error
 	stateConf := &resource.StateChangeConf{
 		Pending:    []string{"Updating", "Creating"},
 		Target:     []string{"Succeeded"},
-		Refresh:    redisStateRefreshFunc(client, resGroup, name),
+		Refresh:    redisStateRefreshFunc(client, ctx, resGroup, name),
```

Comment:
context should be the first argument in these methods (to match conventions in Golang) - since this is minor rather than a blocking issue, I'll push a commit to fix this